ext/pdo_firebird/php_pdo_firebird_int.h

```diff
@@ -83,6 +83,8 @@ typedef struct {
 	char *date_format;
 	char *time_format;
 	char *timestamp_format;
+	
+	unsigned sql_dialect;	
```

Comment:
Would it make sense to make this member a bitfield (not sure about the width), and to take away the respective space from `_reserved`?

Reply:
I thought about it. In principle, 2 bits are now enough to store information about the SQL dialect. However, doubts overcame me here:

```
		H->sql_dialect = PDO_FB_DIALECT;
		if (vars[3].optval) {
			H->sql_dialect = atoi(vars[3].optval);
		}
```

What happens if you pass a number greater than 3? Now such things are simply ignored (the dialect is considered 3). If we will use bit fields, then we need to add an additional check against overflow.

Reply:
As it is now, there is already the possiblity for overflow. `sql_dialect` is declared as `unsigned`, but passed to `isc_dsql_prepare()` which [expects an `unsigned short`](http://docwiki.embarcadero.com/InterBase/XE7/en/Isc_dsql_prepare()). And `atoi()` even returns an `int`, so there might be implementation dependent conversion of negative integer to unsigned. Doesn't look like a real problem, though, since the developer is not supposed to pass some arbitrary value as dialect. Anyhow, I don't really mind if the dialect is stored in a bitfield, or an `unsigned` or `unsigned short`.